### PR TITLE
Updated process name to get job that horizon is processing

### PR DESCRIPTION
**Pull request by qschmick**
Updated the getProcessName method to resolve horizon tasks to the job that it is processing.

That leads to this value
```
horizon:work redis --name=default --supervisor=ip-123-45-678-90-Pspg:supervisor-1 --backoff=0 --max-time=0 --max-jobs=0 --memory=128 --queue=default --sleep=3 --timeout=60 --tries=1
```
becoming this
```
ProcessImportFileJob
```

### CODE REVIEW

File: src/ProcessStamp.php [hunk 1]
```diff
@@ -166,6 +166,24 @@ public static function getProcessName(?string $type = null, ?string $raw_process
...
+                        if (str_ends_with($step['class'], 'CallQueuedHandler')) {
+                            foreach ($step['args'] as $arg) {
+                                if (is_object($arg) && property_exists($arg, 'commandName')) {
+                                    $command_name_parts = explode('\\', $arg->commandName);
+                                    $name = array_pop($command_name_parts);
+                                    break 2;
+                                } elseif (is_array($arg) && array_key_exists('commandName', $arg)) {
+                                    $command_name_parts = explode('\\', $arg['commandName']);
+                                    $name = array_pop($command_name_parts);
+                                    break 2;
```
Comment (solflare) [inline src/ProcessStamp.php:181]: Suuuper minor but what about?
```php
$raw_command_name = $arg?->commandName ?? $arg['commandName'] ?? '';
$parts = explode('\\', $raw_command_name);
$name = array_pop($parts);
```
or we could get it all onto one line 😬:
```php
if ($commandName = array_pop(explode('\\', $arg?->commandName ?? $arg['commandName'] ?? ''))) {
    $name = $commandName;
    break 2;
}
```

**Review by solflare — APPROVED** [review]
Makes sense to me, just a minor comment. Too bad there's not a better way to get the name but this will be much nicer than the current setup 🚀

**Review by tomschlick — APPROVED** [review]

File: src/ProcessStamp.php (hunk 1 above)
Comment (qschmick) [reply src/ProcessStamp.php:181]: @solflare These are some neat ideas. Minor issues with them:
1. Calling a property on an array will cause a warning (`Attempt to read property "commandName" on array`)
2. `array_pop` takes byRef so the second options (1-liners) will throw a warning.

This is part of why I set them up the way I did. I wasn't super happy with it being so verbose but didn't see an easy way to clean up.

Comment (qschmick) [reply src/ProcessStamp.php:177]: @tomschlick @solflare How about this:

```
if ($command_name = json_decode(json_encode($arg), true)['commandName'] ?? null) {
```

Comment (solflare) [reply src/ProcessStamp.php:177]: @qschmick 

>Calling a property on an array will cause a warning

I thought of the same issue and did a test to check, and calling a property on an array with the null coalescing operator doesn't throw an error and properly allows the fallback

>array_pop takes byRef so the second options (1-liners) will throw a warning.

Hmm that's a harder one, and it's surprising there isn't an array function for this that doesn't pass by reference 

Either way the update makes sense but it means we're encoding/decoding for each line of the stack trace until (if) we come across the `commandName` 🤔  Not the biggest deal

Comment (qschmick) [reply src/ProcessStamp.php:177]: @solflare Nice catch with the null coalesce part, missed that in my check. We are only encoding/decoding for the class `CallQueuedHandler` as that is the outer check, then we encode/decode each arg under that class. Should greatly reduce the number of calls.

**Pull request merged**